### DISCUSSION TITLE
[Desktop]The  Input Method（maybe call it ‘typewriter’，some thing to input chinese） do not follow the cursor

### DISCUSSION THREAD

**Discussion by RaySnow** [singerdmx/flutter-quill#525] [Desktop]
My issue is about [Desktop]

The  Input Method（maybe call it ‘typewriter’，some thing to input chinese） do not follow the cursor,  system may not think it's an input. on both MacOs and Windows
it works fine in TextField. 

And, there is another problem, in [Desktop] , only windows, only third pary  Input Method（such as Sogou Input or Baidu Input）.
When I type in chinese after H1, it will change the H1 to normal.
When T type first Enter key after H1, nothing happen, when I type the second Enter key, it will change the H1 to normal, and change the next line to H1.

Please help, thans very much.
 
[video](https://youtu.be/MMLVaENIeBk)

**Comment by Jinlixian** [comment]
Me too
The IME box  does not follow the cursor on MacOS end Windows.
<img width="1327" alt="WX20211216-163111@2x" src="https://user-images.githubusercontent.com/7472208/146335974-01275efd-8168-4afb-9f0e-59f14444d141.png">

**Comment by chifandeyu** [comment]
I upgraded flutter  to 3.3.8, and have the same problem with pinyin input method(微软拼音).

**Comment by chifandeyu** [comment]
> My issue is about [Desktop]
> 
> The Input Method（maybe call it ‘typewriter’，some thing to input chinese） do not follow the cursor, system may not think it's an input. on both MacOs and Windows it works fine in TextField.
> 
> And, there is another problem, in [Desktop] , only windows, only third pary Input Method（such as Sogou Input or Baidu Input）. When I type in chinese after H1, it will change the H1 to normal. When T type first Enter key after H1, nothing happen, when I type the second Enter key, it will change the H1 to normal, and change the next line to H1.
> 
> Please help, thans very much.
> 
> [video](https://youtu.be/MMLVaENIeBk)

excuse me. Have you solved it?

**Comment by demoYang** [comment]
<img width="723" alt="image" src="https://user-images.githubusercontent.com/7134823/209297462-c5c0b8c1-f31a-4888-95fd-fd0fca10c142.png">

**Reply by chifandeyu** [reply]
> <img alt="image" width="723" src="https://user-images.githubusercontent.com/7134823/209297462-c5c0b8c1-f31a-4888-95fd-fd0fca10c142.png">

What does the screenshot mean??

**Reply by demoYang** [reply]
TextField use this function update IME bar,  maybe  flutter_quill  can update IME bar like that  ?  i am not sure

**Comment by foreverpw** [comment]
#1154

**Comment by foreverpw** [comment]
#1159  Fix with scroller
